src/http/client.cc

```diff
@@ -338,39 +338,45 @@ auto client::with_new_connection(Fn&& fn, abort_source* as) {
     });
 }
 
-future<> client::make_request(request req, reply_handler handle, std::optional<reply::status_type> expected) {
-    return do_make_request(std::move(req), std::move(handle), nullptr, std::move(expected));
+future<> client::make_request(request&& req, reply_handler&& handle, std::optional<reply::status_type>&& expected) {
+    return do_with(std::move(req), std::move(handle), [this, expected](request& req, reply_handler& handle) mutable {
+        return do_make_request(req, handle, nullptr, std::move(expected));
+    });
 }
 
-future<> client::make_request(request req, reply_handler handle, abort_source& as, std::optional<reply::status_type> expected) {
-    return do_make_request(std::move(req), std::move(handle), &as, std::move(expected));
+future<> client::make_request(request&& req, reply_handler&& handle, abort_source& as, std::optional<reply::status_type>&& expected) {
+    return do_with(std::move(req), std::move(handle), [this, as{&as}, expected](request& req, reply_handler& handle) mutable {
+        return do_make_request(req, handle, as, std::move(expected));
+    });
 }
 
-future<> client::do_make_request(request req, reply_handler handle, abort_source* as, std::optional<reply::status_type> expected) {
-    return do_with(std::move(req), std::move(handle), [this, as, expected] (request& req, reply_handler& handle) mutable {
-        return with_connection([this, &req, &handle, as, expected] (connection& con) {
-            return do_make_request(con, req, handle, as, expected);
-        }, as).handle_exception_type([this, &req, &handle, as, expected] (const std::system_error& ex) {
-            if (as && as->abort_requested()) {
-                return make_exception_future<>(as->abort_requested_exception_ptr());
-            }
+future<> client::make_request(request& req, reply_handler& handle, abort_source& as, std::optional<reply::status_type> expected) {
+    return do_make_request(req, handle, &as, std::move(expected));
+}
 
-            if (!_retry) {
-                return make_exception_future<>(ex);
-            }
+future<> client::do_make_request(request& req, reply_handler& handle, abort_source* as, std::optional<reply::status_type> expected) {
+    return with_connection([this, &req, &handle, as, expected] (connection& con) {
+        return do_make_request(con, req, handle, as, expected);
+    }, as).handle_exception_type([this, &req, &handle, as, expected] (const std::system_error& ex) {
+        if (as && as->abort_requested()) {
+            return make_exception_future<>(as->abort_requested_exception_ptr());
+        }
 
-            auto code = ex.code().value();
-            if ((code != EPIPE) && (code != ECONNABORTED)) {
-                return make_exception_future<>(ex);
-            }
+        if (!_retry) {
+            return make_exception_future<>(ex);
+        }
 
-            // The 'con' connection may not yet be freed, so the total connection
```

Comment:
There's a lot of indentation fixes here. When this happens, we ask to split the patch into two -- first, that does what it needs, but doesn't re-indent unaffected lines. The second patch restores the indentation. Please, do

Reply:
Sure. Done

Reply:
agggr, squashed, lemme fix that

Reply:
done